kolena/_experimental/object_detection/__init__.py

```diff
@@ -45,5 +47,7 @@
     "TestSuiteMetrics",
     "ThresholdConfiguration",
     "ObjectDetectionEvaluator",
+    "iter_object_detection_results",
+    "create_object_detection_results",
```

Comment:
Are these needed as part of this IoU PR? I'm a bit concerned about expanding the client surface area with these `iter` and `create` functions (especially `create`, since it seems to be just a surface level wrapper around `iter`).

Reply:
Discussed offline, will expose `compute_object_detection_results` returning a `DataFrame` and keep `_iter_object_detection_results` returning an `Iterator[DataFrame]` as a private function for consistency with how we do this elsewhere (e.g. `download_dataset`)